### PR TITLE
chore: Update signTransaction to v5 SDK

### PR DESCRIPTION
**Pull request by arcoraven**
## PR-Codex overview
This PR focuses on enhancing the transaction signing functionality in the backend, introducing support for various transaction types, and improving the related schemas and utility functions.

### Detailed summary
- Removed `.env.test.example` file.
- Updated `ENGINE_URL` in the environment configuration.
- Modified description in `AddressSchema`.
- Introduced `maybeInt` function in `primitiveTypes.ts`.
- Changed transaction type handling in `getReceipt.ts` and `sdk.ts`.
- Added support for new transaction types (eip-4844, eip-7702) in `signTransaction.ts`.
- Implemented tests for different transaction types in `sign-transaction.test.ts`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/utils/sdk.ts [hunk 1]
```diff
@@ -18,14 +18,23 @@ export const thirdwebClient = createThirdwebClient({
  * Helper functions to handle v4 -> v5 SDK migration.
  */
 
-export const toTransactionStatus = (status: "success" | "reverted"): number =>
+export const fromTransactionStatus = (status: "success" | "reverted") =>
```
Comment (arcoraven) [inline src/utils/sdk.ts:21]: Renaming this. `transactionStatus` should refer to the v5 shape, so "from" means v5 -> v4 shape and "to" means v4 -> v5 shape.

File: src/scripts/generate-sdk.ts [hunk 2]
```diff
@@ -2,8 +2,8 @@ import { execSync } from "node:child_process";
 import fs from "node:fs";
 import { kill } from "node:process";
 
-// requires engine to be running locally
-const ENGINE_OPENAPI_URL = "http://localhost:3005/json";
+// Requires Engine to be running locally
+const ENGINE_OPENAPI_URL = "https://demo.web3api.thirdweb.com/json";
```
Comment (joaquim-verges) [inline src/scripts/generate-sdk.ts:6]: is this up to date now?

File: src/server/routes/backend-wallet/signTransaction.ts [hunk 3]
```diff
@@ -52,16 +59,49 @@ export async function signTransaction(fastify: FastifyInstance) {
...
+          type: transaction.type
+            ? toTransactionType(transaction.type)
+            : undefined,
+          accessList: transaction.accessList,
+          maxFeePerGas: maybeBigInt(transaction.maxFeePerGas),
+          maxPriorityFeePerGas: maybeBigInt(transaction.maxPriorityFeePerGas),
+          ccipReadEnabled: transaction.ccipReadEnabled,
+        };
+      } catch (e) {
+        throw createCustomError(
```
Comment (joaquim-verges) [inline src/server/routes/backend-wallet/signTransaction.ts:94]: what throws in here? only see safe stuff unless im missing something

File: src/server/routes/backend-wallet/signTransaction.ts [hunk 4]
```diff
@@ -52,16 +59,49 @@ export async function signTransaction(fastify: FastifyInstance) {
...
+          'This backend wallet does not support "signTransaction".',
+          StatusCodes.BAD_REQUEST,
+          "SIGN_TRANSACTION_UNIMPLEMENTED",
+        );
+      }
 
-      const signer = await wallet.getSigner();
-      const signedMessage = await signer.signTransaction(transaction);
+      // @TODO: Assert type to viem TransactionSerializable.
+      let serializableTransaction: any;
```
Comment (joaquim-verges) [inline src/server/routes/backend-wallet/signTransaction.ts:75]: we have a exported type for this: `SerializableTransaction`

File: test/e2e/tests/sign-transaction.test.ts [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+            ],
+          },
+        ],
+        customData: {
+          someCustomField: "customValue",
+        },
+      },
+    });
+
+    expect(res.result).toEqual(
```
Comment (joaquim-verges) [inline test/e2e/tests/sign-transaction.test.ts:137]: v nice

**Review by AsimAlam — COMMENTED** [review]
Nice work

**Review by joaquim-verges — APPROVED** [review]

**Pull request merged**